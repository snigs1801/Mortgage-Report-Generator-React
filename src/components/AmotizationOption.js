import React from "react";

// radio button component
const AmotizationOption = ({ mortgageTerm, targetValue, setMortgageTerm }) => {
  return (
    <label
      className={mortgageTerm === Number(targetValue) ? "selected-input" : null}
    >
      <input
        type="radio"
        name="mortgageTerm"
        value={targetValue}
        onChange={e => setMortgageTerm(Number(e.target.value))}
      />
      {targetValue}
    </label>
  );
};

export default AmotizationOption;
