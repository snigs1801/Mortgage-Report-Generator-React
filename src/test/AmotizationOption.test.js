import React from 'react';import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import AmotizationOption from "../components/AmotizationOption"; 


Enzyme.configure({ adapter: new Adapter() });

describe('AmotizationOption component Test Cases', () => {
  const container = shallow(<AmotizationOption />);
  
  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot();
  });

  it('should have a input radio', () => { 
    expect(container.find('input[type="radio"]').length).toEqual(1);
  });
});