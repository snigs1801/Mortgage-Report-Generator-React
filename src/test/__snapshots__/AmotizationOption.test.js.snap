// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmotizationOption component Test Cases should match the snapshot 1`] = `"<label><input type=\\"radio\\" name=\\"mortgageTerm\\"/></label>"`;
